File: server/src/main/java/cn/iinti/atom/system/LoginRequired.java
package cn.iinti.atom.system;


import proguard.annotation.Keep;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
@Keep
public @interface LoginRequired {
    boolean forAdmin() default false;

    boolean apiToken() default false;

    /**
     * 标记了alert的接口发生调用时，系统发出敏感接口操作事件
     */
    boolean alert() default false;
}

File: server/src/main/kotlin/cn/iinti/atom/system/LoginRequired.kt
package cn.iinti.atom.system

import kotlin.annotation.AnnotationRetention.RUNTIME
import kotlin.annotation.AnnotationTarget.FUNCTION
import kotlin.annotation.AnnotationTarget.CLASS
import kotlin.annotation.AnnotationTarget.ANNOTATION_CLASS
import kotlin.annotation.AnnotationTarget.CONSTRUCTOR
import kotlin.annotation.AnnotationTarget.PROPERTY_GETTER
import kotlin.annotation.AnnotationTarget.PROPERTY_SETTER
import kotlin.reflect.KClass

@Retention(AnnotationRetention.RUNTIME)
@Target(CLASS, FUNCTION, ANNOTATION_CLASS, CONSTRUCTOR, PROPERTY_GETTER, PROPERTY_SETTER)
annotation class LoginRequired(
    val forAdmin: Boolean = false,
    val apiToken: Boolean = false,
    val skipLogRecord: Boolean = false,
    val alert: Boolean = false,
    val value: String = ""
)